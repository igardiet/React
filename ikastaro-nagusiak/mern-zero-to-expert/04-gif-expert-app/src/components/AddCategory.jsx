import { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({ onNewCategory }) => {
  const [inputValue, setInputValue] = useState('');

  const handleInputChange = ({ target }) => setInputValue(target.value);

  const onSubmit = (event) => {
    event.preventDefault();
    if (inputValue.trim().length <= 0) return;
    // setCategories((categories) => [inputValue, ...categories]);
    onNewCategory(inputValue.trim());
    setInputValue('');
  };

  return (
    <form onSubmit={onSubmit} aria-label='form'>
      <input
        type='text'
        placeholder='Search Gifs'
        value={inputValue}
        onChange={handleInputChange}
      />
    </form>
  );
};

AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired,
};
