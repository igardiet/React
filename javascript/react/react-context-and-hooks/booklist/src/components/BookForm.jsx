
export const BookForm = () => {
    const {addBook} = useContext(BookContext)
    const [title, setTitle] = useState('')
    const [author, setAuthor] = useState('')
    const handleSubmit = (e) => {
        e.preventDefault()
        addBook(title, author)
        setTitle('')
        setAuthor('')
    }

  return (
    <div>BookForm</div>
  )
}
