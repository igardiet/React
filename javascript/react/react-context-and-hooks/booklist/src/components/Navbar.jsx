import { useContext } from 'react';
import { BookContext } from '../contexts/BookContext';

export const Navbar = () => {
  const { books } = useContext(BookContext);
  return (
    <div className='navbar'>
      <h1>Book List</h1>
      <p>Currently you have {books.length} books to get through...</p>
    </div>
  );
};
