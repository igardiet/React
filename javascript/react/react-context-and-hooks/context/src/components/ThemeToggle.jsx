import {
  // Component,
  useContext,
} from 'react';
import { ThemeContext } from '../contexts/ThemeContext';

// OLD CLASS COMPONENT VERSION
// export default class ThemeToggle extends Component {
//   static contextType = ThemeContext;
//   render() {
//     const { theme } = this.context;
//   }
// }

// USECONTEXT INSIDE FUNCTIONAL COMPONENTS
export const ThemeToggle = () => {
  const { theme } = useContext(ThemeContext);
  return <button onClick={theme}>Theme toggle</button>;
};
