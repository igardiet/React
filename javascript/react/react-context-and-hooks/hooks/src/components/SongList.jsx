import { useState } from 'react';
import { v4 as uuidv4 } from 'uuid';

export const SongList = () => {
  const [songs, setSongs] = useState([
    { title: 'song one', id: 1 },
    { title: 'song two', id: 2 },
    { title: 'song three', id: 3 },
  ]);
  const addSong = () => {
    setSongs([...songs, { title: 'new song', id: uuidv4() }]);
  };
  return (
    <div className='song-list'>
      <ul>
        {songs.map((song) => {
          return <li key={song.id}>{song.title}</li>;
        })}
      </ul>
      <button onClick={addSong}>Add song!</button>
    </div>
  );
};
