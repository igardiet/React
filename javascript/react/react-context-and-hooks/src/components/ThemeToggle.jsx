import { Component } from 'react';
import { ThemeContext } from '../contexts/ThemeContext';

export default class ThemeToggle extends Component {
  static contextType = ThemeContext;
  render() {
    const { theme } = this.context;
    return <button onClick={theme}>Theme toggle</button>;
  }
}
