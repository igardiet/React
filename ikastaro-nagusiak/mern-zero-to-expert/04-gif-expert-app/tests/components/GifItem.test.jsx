/**
 * @jest-environment jsdom
 */

import { render } from '@testing-library/react';
import { GifItem } from '../../src/components/GifItem';

describe('Tests in <GifItem />', () => {
  const title = 'Naruto';
  const url = 'https://naruto.com/naruto.jpg';

  test('Must match with snapshot', () => {
    const { container } = render(<GifItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });
});
