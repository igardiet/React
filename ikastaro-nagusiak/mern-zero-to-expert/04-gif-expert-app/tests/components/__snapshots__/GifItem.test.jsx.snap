// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <GifItem /> Must match with snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Naruto"
      src="https://naruto.com/naruto.jpg"
    />
    <p>
      Naruto
    </p>
  </div>
</div>
`;
